From 80ef4b257c8b85dd5f8e15bb64b59ccf28131e7b Mon Sep 17 00:00:00 2001
From: Nicolas Boichat <drinkcat@google.com>
Date: Tue, 31 May 2016 19:05:05 +0800
Subject: [PATCH] egl/dri2: dri2_make_current: Set EGL error if bindContext
 fails

Setting error to EGL_BAD_MATCH is what CTS tests expect.

BUG=chrome-os-partner:53869
TEST=dEQP-EGL.functional.surfaceless_context#rgb565_no_depth_no_stencil
     passes.

Change-Id: I2944cef5ab3912cbd01e6bdedd0a05966d57bb22
---
 src/egl/drivers/dri2/egl_dri2.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/egl/drivers/dri2/egl_dri2.c b/src/egl/drivers/dri2/egl_dri2.c
index 564ad61..e75940a 100644
--- a/src/egl/drivers/dri2/egl_dri2.c
+++ b/src/egl/drivers/dri2/egl_dri2.c
@@ -1263,6 +1263,7 @@ dri2_make_current(_EGLDriver *drv, _EGLDisplay *disp, _EGLSurface *dsurf,
       _eglPutSurface(old_rsurf);
       _eglPutContext(old_ctx);
 
+      _eglError(EGL_BAD_MATCH, "bindContext error");
       return EGL_FALSE;
    }
 }
-- 
2.8.0.rc3.226.g39d4020

